src/blockdata/witness.rs

```diff
@@ -180,6 +180,25 @@ impl Witness {
             .expect("writers on vec don't error, space granted through previous resize");
         self.content[end_varint..].copy_from_slice(new_element);
     }
+
+    /// Gets nth element from the witness stack. Does not allocate
+    pub fn get(&self, index: usize) -> Option<&[u8]> {
```

Comment:
Why not to keep the whole map of indexes as a part of the struct itself? This will make indexing performance much better.

Indexes are needed for circumstances like "get the previous to last item in witness stack" useful in Taproot witness parsing (for instance "if there is more than two stack elems after annex removal, get the second to the last as a script")